src/prepare.ts

```diff
@@ -150,12 +156,16 @@ async function prepareElement(
         { ...memoElement, type: memoElement.type.type },
         errorHandler,
         context,
+        dispatcher,
       );
     }
     case ELEMENT_TYPE.FUNCTION_COMPONENT: {
       const functionElement = element as FunctionComponentElement<unknown>;
-      setDispatcherContext(context);
-      return [functionElement.type(functionElement.props), context];
+      setDispatcherContext(dispatcher, context);
+      registerDispatcher(dispatcher);
+      const children = functionElement.type(functionElement.props);
+      await Promise.all(popAwaitImmediatelyPromises(dispatcher));
+      return [children, context];
```

Comment:
Is `children` more than one `ReactNode` here?
Considering the function output signature of `Promise<[ReactNode, PrepareContext, Promise<void>?]> `

Reply:
A `ReactNode` could be either a single react node or an array of nodes. This is why we use `React.Children.toArray` below to make sure that we are dealing with an array of elements.
https://reactnative.dev/docs/react-node